quarkus/runtime/src/test/java/or/keycloak/quarkus/runtime/cli/PicocliTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package or.keycloak.quarkus.runtime.cli;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.keycloak.quarkus.runtime.cli.Picocli;
+import org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand;
+import org.keycloak.quarkus.runtime.configuration.ConfigArgsConfigSource;
+import org.keycloak.quarkus.runtime.configuration.test.AbstractConfigurationTest;
+
+import io.smallrye.config.SmallRyeConfig;
+import picocli.CommandLine;
+import picocli.CommandLine.Help;
+
+public class PicocliTest extends AbstractConfigurationTest {
+
+    // TODO: could utilize CLIResult
+    private class NonRunningPicocli extends Picocli {
```

Comment:
Follow-up: It should be separated from the specific test cases + polished. Might be good also provide some annotations(`@NonRunningServer`) + leverage @Launch annotation for it. @shawkins WDYT?

Reply:
Yes, it can be refined as needed. As for mixing with the Launch annotation, it's possible, but there can be mismatches (such as capturing the console coloring) where it's not transparently the same. The goal would be to keep it runnable directly from the runtime module.